src/math/trig.c

```diff
@@ -3,574 +3,100 @@
-	2103610053,
-	2104925108,
-	2106220351,
-	2107495769,
-	2108751351,
-	2109987084,
-	2111202958,
-	2112398959,
-	2113575079,
-	2114731304,
-	2115867625,
-	2116984030,
-	2118080510,
-	2119157053,
-	2120213650,
-	2121250291,
-	2122266966,
-	2123263665,
-	2124240379,
-	2125197099,
-	2126133816,
-	2127050521,
-	2127947205,
-	2128823861,
-	2129680479,
-	2130517051,
-	2131333571,
-	2132130029,
-	2132906419,
-	2133662733,
-	2134398965,
-	2135115106,
-	2135811152,
-	2136487094,
-	2137142926,
-	2137778643,
-	2138394239,
-	2138989707,
-	2139565042,
-	2140120239,
-	2140655292,
-	2141170196,
-	2141664947,
-	2142139540,
-	2142593970,
-	2143028233,
-	2143442325,
-	2143836243,
-	2144209981,
-	2144563538,
-	2144896909,
-	2145210091,
-	2145503082,
-	2145775879,
-	2146028479,
-	2146260880,
-	2146473079,
-	2146665075,
-	2146836865,
-	2146988449,
-	2147119824,
-	2147230990,
-	2147321945,
-	2147392689,
-	2147443221,
-	2147473541,
-	2147483647
-};
+/* Compute fixed point cordicsine with table lookup and interpolation*/
+/*	 The cordic sine algorithm converges, when the angle is in the range*/
+/*	 [-pi/2, pi/2).If an angle is outside of this range, then a multiple of	*/
+/*	 pi/2 is added or subtracted from the angle until it is within the range*/
+/*	 [-pi/2,pi/2).Start with the angle in the range [-2*pi, 2*pi) and output*/
+/*	 has range in [-1.0 to 1.0]	*/
+/*	+------------------+-----------------+--------+--------+*/
+/*	| thRadFxp		   | cdcsinth		 |thRadFxp|cdcsinth|*/
+/*	+----+-----+-------+----+----+-------+--------+--------+*/
+/*	|WLen| FLen|Signbit|WLen|FLen|Signbit| Qformat| Qformat|*/
+/*	+----+-----+-------+----+----+-------+--------+--------+*/
+/*	| 32 | 28  |  1	   | 32	| 30 |	1	 | 4.28	  | 2.30   |*/
+/*	+------------------+-----------------+--------+--------+*/
 
-/* Sine lookup table read */
-static inline int32_t sine_lookup(int idx)
+int32_t sin_fixed(int32_t th_rad_fxp)
 {
-	int32_t s;
-	int i1;
+	/*cordic_atan2_lookup_table = atan(2.^-(0:N-1)) N = 31	*/
+	/*CORDIC Gain is cordic_gain = prod(sqrt(1 + 2.^(-2*(0:31-1)))) */
+	/*Inverse CORDIC Gain,inverse_cordic_gain = 1 / cordic_gain	 */
+	static const int32_t inpLUT[31] = { 843314857, 497837829, 263043837, 133525159,
+	 67021687, 33543516, 16775851, 8388437, 4194283, 2097149, 1048576, 524288,
+	 262144, 131072, 65536, 32768, 16384, 8192, 4096, 2048, 1024, 512, 256, 128,
+	 64, 32, 16, 8, 4, 2, 1 };
+	int32_t b_idx;
+	int32_t b_yn;
+	int32_t xn;
+	int32_t xtmp;
+	int32_t ytmp;
+	int32_t cdc_sin_th;
+	int32_t negate;
+   /* 421657428 , deg = 90.000000 */
+	const int32_t CORDICSINE_PIOVERTWO_Q28FL	= Q_CONVERT_FLOAT(1.570796325802803, 28);
+   /* 843314857,  deg = 90.000000  */
+	const int32_t CORDICSINE_PIOVERTWO_Q29FL	= Q_CONVERT_FLOAT(1.570796327665448, 29);
+   /* 1686629713, deg = 90.000000  */
+	const int32_t CORDICSINE_PIOVERTWO_Q30FL	= Q_CONVERT_FLOAT(1.570796326734126, 30);
+   /* 652032874 , deg = 69.586061*/
+	const int32_t CORDICSINE_LUT_Q29FL =	 Q_CONVERT_FLOAT(1.214505869895220, 29);
```

Comment:
best to have a newline here after the last variable declaration. Btw, the indentation looks a bit odd above ?

Reply:
Indentation looks odd due to diff/Github.